src/components/SearchBar.tsx

```diff
@@ -57,18 +57,30 @@ export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
     clearInput() {
         let searchInput: any = document.getElementById("SearchInputText");
         searchInput.value = '';
-        this.setState({ hasText: false });
+        this.setState({
+            hasText: false,
+            expanded: false // collapse filter options menu when text is cleared
+        });
```

Comment:
Hmm...when there is no search text, `expanded` is always false. However, when there is text on the search bar, `expanded` can be true or false based on user click.